### ISSUE TITLE
Rails 5.1 support

### ISSUE THREAD

**Issue by zachaysan** [take-five/activerecord-hierarchical_query#22]
Hi there,

Thank you for making this. I know you do not have a lot of time so I've created a branch for Rails 5.1 support and I've placed it here:

https://github.com/zachaysan/activerecord-hierarchical_query/tree/rails-5.1

If you would like me to created a pull request you can either add me to the repo and I can create the same branch for you or you can create the branch yourself and I'll issue the pull request. It's up to you.

All the tests pass after updating some gems and I verified that basic features work with my rails project. I try to keep my rails projects as up to date as possible so I will continue to upgrade this gem as I need it.

For other using rails-5.1 the following code in your Gemfile will do the trick as a temporary measure:

```
# TODO: Check back at https://github.com/take-five/activerecord-hierarchical_query to replace gem
gem "activerecord-hierarchical_query",
    ">= 1.0.0",
    git: "https://github.com/zachaysan/activerecord-hierarchical_query",
    branch: "rails-5.1"
```

**Comment by take-five** [comment]
Hi @zachaysan 

Thank you for contributing! Unfortunately I don't really have time to maintain this library anymore, but I'm happy to accept pull requests from the community.

As of your work to make it compatible with rails 5.1. I looked briefly at your commits and it seems that we only needed to update gemspec. In this case I suggest you to open a pull request to branch `rails-5` which I'll happily merge and push new version to rubygems. Perhaps README will also require a minor update.

**Comment by zachaysan** [comment]
Sounds great, I'll do so when I'm back at a computer! I'll check the README
as well.

**Comment by zachaysan** [comment]
Ok, I filed two pull requests. I added a note in the README for people to tag me if there are issues with Rails 5 since I'll need to keep it working for my own project anyway! Have a great day :smile:

**Comment by take-five** [comment]
Merged pull requests and released version 1.0.1
